File: src/main/java/gc/array/RemoveElement.java
package gc.array;

import org.junit.Test;

public class RemoveElement {
	public int removeElement(int[] A, int elem) {
		int removed = 0;
		for(int i=0;i<A.length;i++){
			if(A[i]== elem){
				removed++;
			}else {
				A[i-removed] = A[i];
			}
		}
		return A.length - removed;
	}
	
	@Test
	public void test(){
		int[] a = new int[]{1,2,3,4,51,2,3,45,23,2};
		int i = removeElement(a, 2);
		System.out.println(i);
	}
}

File: src/main/scala/array/RemoveElement.scala
package array

object RemoveElement {
  def removeElement(nums: Array[Int], tt: Int): Int = {
    var removed = 0
    for (i <- 0 until nums.length) {
      if (nums(i) == tt) {
        removed += 1
      } else {
        nums(i - removed) = nums(i)
      }
    }
    nums.length - removed
  }

}
